### PR TITLE
[exporter/datasetexporter] Various improvements and fixes for the DataSet Log Exporter plugin

### PR DESCRIPTION
**Pull request by tomaz-s1**
# Related issue
#20660

## Remove OtelExporter - Log - prefix from all the DataSet event message fields

This prefix was added in the past for testing purposes, but it's not desired.

## Correctly handle LogRecord severity

The code has been updated to correctly map ``LogRecord.SeverityNumber`` and ``LogRecord.SeverityText`` field value to the DataSet event log event severity (https://app.scalyr.com/help/parsing-logs#specialAttrs, https://github.com/scalyr/logstash-output-scalyr/blob/master/lib/logstash/outputs/scalyr.rb#L70)

Redundant `severity.number` and `severity.text` attribute has also been removed from DataSet event.

## Remove redundant flags and flags.is_sampled attribute

I removed redundant `flags` and `flags.is_sampled` attribute from the DataSet events. DataSet doesn't do anything with those fields and they are also redundant - if event is ingested into DataSet and displayed in the UI this means it has already been sampled so `flags.is_sampled` tells us no new information.

### CODE REVIEW

File: .chloggen/export-logs-resource-info-based-configuration.yaml
Comment (tomaz-s1) [inline .chloggen/export-logs-resource-info-based-configuration.yaml:12]: This is a left over from another PR from @zdaratom which for some reason was missing from his PR.

**Comment by tomaz-s1** [comment]
Not 100% sure, but the failure here (https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/5375815914/jobs/9752150013?pr=23672) also doesn't seem to be related to my changes.

I would re-run it to see if it's a flakey tests, but I don't have permission to do that.

**Comment by dmitryax** [comment]
@tomaz-s1 the failing integration test was fixed in the main branch. Please rebase

**Comment by tomaz-s1** [comment]
@dmitryax Thanks. Done. All the checks appear to be passing now.

**Comment by tomaz-s1** [comment]
@dmitryax When you get a chance, please let me know if there is anything else which needs to be done on my side. Thanks a lot!

**Comment by tomaz-s1** [comment]
@atoulme As a code owner, could you please review when you get a chance? Thanks.

File: exporter/datasetexporter/logs_exporter.go
Comment (atoulme) [inline exporter/datasetexporter/logs_exporter.go:104]: Otel offers probabilistic head sampling for logs and traces, you can use that. There is no need for a TODO entry here as this is best captured with an issue on the github repository as a follow up instead.

File: .chloggen/export-logs-resource-info-based-configuration.yaml
Comment (atoulme) [reply .chloggen/export-logs-resource-info-based-configuration.yaml:12]: not sure I follow, what is this changelog entry doing here?

**Comment by atoulme** [comment]
OK, fwiw it's much easier if you run 4 PRs instead of one. Please do peel off at least the metadata changes as a separate `[chore]` PR that we can easily merge. I can't judge the changes much except for the TODO entry, see my comment. Please address my comment and I will approve the PR.

File: .chloggen/export-logs-resource-info-based-configuration.yaml
Comment (tomaz-s1) [reply .chloggen/export-logs-resource-info-based-configuration.yaml:12]: We missed / forgot to add changelog as part of that PR (https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/23250), so I added it here.

File: exporter/datasetexporter/logs_exporter.go
Comment (tomaz-s1) [reply exporter/datasetexporter/logs_exporter.go:104]: Per my reading and understanding of this sampling processor (https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/probabilisticsamplerprocessor), the processor is not flexible enough to only sample specific log messages.

So I could define a processor rule to sample all the log exporter logging messages, but not just this specific log message.

Other log messages are not that noisy / verbose so I would only want to sample this specific log message.

And I don't really want to use debug log level here either. I still want to log it, ideally, under warning log level, but only emit one message every X seconds in case there are lot of messages like that to prevent noise, overhead, etc.

Please correct me if I'm wrong / misunderstanding it.

Or is there some other existing code pattern / API in the code base I could use to do that?

**Comment by tomaz-s1** [comment]
@atoulme Thanks for a review.

And yeah, I agree - it would make it a bit easier to review if those were 4 separate PRs, but per PR description above, I tried first with combining all 4 changes together first since they are small, related and makes it a bit easier for us. 

But in case it's needed, I can definitely split it out :)

Having said that, I will split out metadata change in a separate PR.

**Comment by tomaz-s1** [comment]
Moved metadata change in a new and separate PR - https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/23885.

**Review by martin-majlis-s1 — APPROVED** [review]

File: exporter/datasetexporter/logs_exporter.go
Comment (tomaz-s1) [reply exporter/datasetexporter/logs_exporter.go:104]: If not, to unblock the PR, I'm also fine with removing this TODO comment for now and open an issue instead.

File: .chloggen/export-logs-resource-info-based-configuration.yaml
Comment (atoulme) [reply .chloggen/export-logs-resource-info-based-configuration.yaml:12]: Please create a chore pr to add the changelog asap

Comment (atoulme) [reply .chloggen/export-logs-resource-info-based-configuration.yaml:12]: I see the changelog entry there: .chloggen/export-logs-resource-info-based-configuration.yaml so I don’t think it was missed.

File: exporter/datasetexporter/logs_exporter.go
Comment (atoulme) [reply exporter/datasetexporter/logs_exporter.go:104]: Please do.

File: .chloggen/export-logs-resource-info-based-configuration.yaml
Comment (tomaz-s1) [reply .chloggen/export-logs-resource-info-based-configuration.yaml:12]: @atoulme I'm not sure I understand your comment. I don't see that changelog file in the main branch (https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/.chloggen), I only see it in this one where I added it.

Am I missing something? Thanks.

**Comment by tomaz-s1** [comment]
Integration tests failure doesn't seem to be related to my changes. Will sync this branch again with latest main which will trigger another build, to see if it helps.

**Comment by tomaz-s1** [comment]
@atoulme I removed the TODO comments / annotations and synced up this branch with latest main. It looks like everything is passing now. Can you please check again when you get a chance? Thanks.

File: .chloggen/export-logs-resource-info-based-configuration.yaml
Comment (tomaz-s1) [reply .chloggen/export-logs-resource-info-based-configuration.yaml]: I went ahead and removed that changelo file.

**Review by atoulme — APPROVED** [review]
LGTM

**Pull request merged**
